### PR TITLE
Ros2 devel (simulation_ws)

### PR DESCRIPTION
**Pull request by swatifulzele**
*Issue #, if available:*

*Description of changes:*
1. Migrated CMakeLists.txt and package.xml from ros1 to ros2.
2. Migrated spawn_turtlebot.launch (ros1) to spawn_turtlebot.launch.py (ros2).
3. Migrated urdf files from ros1 to ros2.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,34 +21,15 @@ env:
   - GH_USER_NAME="travis-ci"
   - GH_USER_EMAIL="travis@travis-ci.org"
   - AWS_DEFAULT_REGION=us-west-2
+  - WORKSPACES="simulation_ws"
```
Comment (AAlon) [inline .travis.yml]: Already specified WORKSPACES in the matrix above, no need to add a global definition

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +1,65 @@
...
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+################################################################################
+# Find ament packages and libraries for ament and system dependencies
+################################################################################
```
Comment (AAlon) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Why do we need all of these? Where do we use any of these besides ament_cmake in this package?

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt [hunk 3]
```diff
@@ -1,13 +1,65 @@
...
+# Set minimum required version of cmake, project name and compile options
+################################################################################
+cmake_minimum_required(VERSION 3.5)
 project(turtlebot3_description_reduced_mesh)
 
-find_package(catkin REQUIRED COMPONENTS
-  gazebo_ros
-)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
```
Comment (AAlon) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: If this package doesn't build C++/C it shouldn't set CMAKE_CXX_STANDARD or add_compile_options.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt [hunk 4]
```diff
@@ -1,13 +1,65 @@
...
+# Setup for python modules and scripts
+################################################################################
+
+################################################################################
+# Declare ROS messages, services and actions
+################################################################################
 
-catkin_package(DEPENDS gazebo_ros)
+################################################################################
+# Declare ROS dynamic reconfigure parameters
```
Comment (AAlon) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Need comments cleanup. Most of the comments are either obsolete or not really useful

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+# INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+# PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""Launch turtlebot3_description_reduced_mesh and a rotate node."""
+
+import tempfile
```
Comment (mm318) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: This import is never used.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import subprocess
+import os
+import sys
+import xacro
```
Comment (mm318) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: I don't believe this import is ever used either. `xacro` is called via a subprocess, not via a Python API.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import lifecycle_msgs.msg
```
Comment (mm318) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: I don't see this import being used either.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+import launch
+import launch_ros.actions
```
Comment (mm318) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: `launch.actions` is being used, I don't think `launch_ros.actions` is being used.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+def generate_launch_description():
+    """Main."""
+    use_sim_time_false = launch.actions.DeclareLaunchArgument(
+        'use_sim_time',
+        default_value='true',
+        description='Use sim time',
+        )
+
+    turtlebot3_model_waffle_pi = os.environ.get('TURTLEBOT3_MODEL', 'waffle_pi')
+    #turtlebot3_model_waffle_pi = 'waffle_pi'
```
Comment (mm318) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: Remove commented code.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/package.xml [hunk 10]
```diff
@@ -1,15 +1,25 @@
...
-  <maintainer email="ros-contributions@amazon.com">AWS RoboMaker</maintainer>
-  <buildtool_depend>catkin</buildtool_depend>
+  <author email="aws-robomaker@amazon.com">AWS RoboMaker</author>
+  <maintainer email="aws-robomaker@amazon.com">AWS RoboMaker</maintainer>
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <depend>rclcpp</depend>
+  <depend>gazebo_ros_pkgs</depend>
   <depend>gazebo_ros</depend>
   <depend>gazebo_plugins</depend>
+  <depend>ros2launch</depend>
```
Comment (mm318) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/package.xml]: This should probably be
```
  <depend>launch</depend>
  <depend>launch_ros</depend>
```
instead.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py (hunk 5 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: ok.. removed "tempfile"

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py (hunk 6 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: Yes.."removed import xacro"

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py (hunk 7 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: Ya.. removed

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py (hunk 8 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: Yes...changed to launch.actions

File: simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py (hunk 9 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/launch/spawn_turtlebot.launch.py]: removed the commented code

File: simulation_ws/src/turtlebot3_description_reduced_mesh/package.xml (hunk 10 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/package.xml]: Yes.. replaced

**Comment by swatifulzele** [comment]
@mm318 
I have done with changes suggested by you on this PR.
Though I haven't done any major changes but still CI build is failing here.
I don't see any error in CI build logs. So not able to figure it out.
Could you please help me to fix CI build issue ?

**Comment by mm318** [comment]
@swatifulzele We have noticed the same issue, where Travis CI reports failure even though the log says `Done. Your build exited with 0.`

We are still trying to figure out the solution. Thanks.

**Review by AAlon — CHANGES_REQUESTED** [review]
Thanks Swati, some comments below.

**Comment by AAlon** [comment]
I see that most packages still have `COLCON_IGNORE` - I think it makes sense to continue working on this PR and only merge once it is functional (rather than posting new PRs), does that make sense @swatifulzele? please let us know if there's anything blocking you.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt (hunk 2 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Yes, removed.

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt (hunk 3 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: removed

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt (hunk 4 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Yes, done.

**Comment by swatifulzele** [comment]
> I see that most packages still have `COLCON_IGNORE` - I think it makes sense to continue working on this PR and only merge once it is functional (rather than posting new PRs), does that make sense @swatifulzele? please let us know if there's anything blocking you.

Added remaining packages so COLCON_IGNORE is removed.

**Comment by swatifulzele** [comment]
> _Issue #, if available:_
> 
> _Description of changes:_
> 
>     1. Migrated CMakeLists.txt and package.xml from ros1 to ros2.
> 
>     2. Migrated spawn_turtlebot.launch (ros1) to spawn_turtlebot.launch.py (ros2).
> 
>     3. Migrated urdf files from ros1 to ros2.
> 
> 
> By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
1. Changes done in turtlebot3_reduced_mesh in launch files and in CMakeLists as per comment on PR.
2. Ported  person_detection_simulation and aws_robomaker_simulation_common from ros1 to ros2.
3. [Optional] Modify config yaml files for Kinesis, Rekognition and Cloudwatch.
4. If Rekognition is Enabled, Follow Steps to start Rekognition [ReadMe_Rekognization_steps.txt](https://github.com/aws-robotics/aws-robomaker-sample-application-persondetection/files/3690590/ReadMe_Rekognization_steps.txt)

**Steps to execute launch files:**
$ Copy aws-robomaker-small-house-world package to aws-robomaker-sample-application-persondetection/simulation_ws/src/deps/ 
$ source /opt/ros/dashing/setup.bash
$ export TURTLEBOT3_MODEL=waffle_pi
$ source /usr/share/gazebo/setup.sh
$ colcon build
$ source install.setup.sh
$ ros2 launch person_detection_simulation small_house.launch.py gui:=true
OR
$ ros2 launch person_detection_simulation small_house_turtlebot_navigation.launch.py gui:=true

**Note:** 
Currently map is taken from $HOME due to string limit on the path (ros2 limitation).

**Observation:**
Sometimes it is observed that turtlebot model does not move along with navigation path in rviz. However, it navigates in gazebo.
------------------------------------------------------------------------------------------------------------------------------
1. Removed `turtlebot3-description-reduced-mesh` from `aws-robomaker-sample-application-persondetection/simulation_ws/src` and added it in .rosinstall.
2. Placed param in person_detection_simulation. 
3. In person_detection_simulation, added **small_house_turtlebot_navigation.launch.py**, **turtlebot3_navigation.launch.py** -> this needs map_path(map.yaml and map.pgm is placed in `aws_robomaker_small_house_world` ) and param (set initial pose in yaml file which application decides) for navigation node. 
4. To able to run the application, temporarily map.pgm and map.yaml is copied to **tmp/maps** from   **aws_robomaker_small_house_world/maps/turtlebot3_waffle_pi/***. This change can be reverted back once the ros2 issue is resolved.
5. We have tested it on both, locally and on robomaker and it runs successfully. To run this application on robomaker follow steps mentioned here
[ReadMe_Robomaker_Steps.txt](https://github.com/aws-robotics/aws-robomaker-sample-application-persondetection/files/3734097/ReadMe_Robomaker_Steps.txt)
6. I tried to push the latest changes done and faced branch conflict error(as ros2 robot_ws is merged now). I tried to resolve it from my side, however not able to resolve it.  It seems that it need to be resolve from your side.
7.  **Please resolve the branch conflict so that the latest commits get pushed successfully.**

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt [hunk 11]
```diff
@@ -1,13 +1,20 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(turtlebot3_description_reduced_mesh)
```
Comment (AAlon) [inline simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Let's remove turtlebot3_description_reduced_mesh from this repository and use the one in https://github.com/aws-robotics/turtlebot3-description-reduced-mesh (ros2 branch). Add it to the .rosinstall like so: https://github.com/aws-robotics/aws-robomaker-sample-application-voiceinteraction/blob/ros2-devel/simulation_ws/.rosinstall#L2

**Review by AAlon — CHANGES_REQUESTED** [review]

File: simulation_ws/.rosinstall [hunk 12]
```diff
@@ -1,3 +1,2 @@
-- git: {local-name: src/deps/aws-robomaker-small-house-world, uri: "https://github.com/aws-robotics/aws-robomaker-small-house-world", version: v0.0.1}
-- git: {local-name: src/deps/turtlebot3, uri: "https://github.com/ROBOTIS-GIT/turtlebot3", version: d3cdcc6647812ae9a83f05e626cdae322923ac84}
-- git: {local-name: src/deps/turtlebot3_simulations, uri: "https://github.com/ROBOTIS-GIT/turtlebot3_simulations", version: 1.2.0}
+# - git: {local-name: src/deps/aws-robomaker-small-house-world, uri: "https://github.com/aws-robotics/aws-robomaker-small-house-world", version: v0.0.1}
```
Comment (AAlon) [inline simulation_ws/.rosinstall]: small-house is merged, can you please uncomment and use the ros2 branch?

**Review by AAlon — CHANGES_REQUESTED** [review]

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt (hunk 11 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Yes, done

File: simulation_ws/.rosinstall (hunk 12 above)
Comment (swatifulzele) [reply simulation_ws/.rosinstall]: ok, done

File: simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt (hunk 11 above)
Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Removing `turtlebot3_description_reduced_mesh` from this repository led to the changes in the small_house_turtlebot_navigation.launch.py in the` aws-robomaker-small-house-world`. How to merge these changes? Should I need to open an issue or raise PR  for `aws-robomaker-small-house-world` on ros2 branch?

Comment (swatifulzele) [reply simulation_ws/src/turtlebot3_description_reduced_mesh/CMakeLists.txt]: Raised PR on ros2 branch for `aws-robomaker-small-house-world `, please review for the same.

File: simulation_ws/src/person_detection_simulation/config/kvs_config.yaml [hunk 13]
```diff
@@ -0,0 +1,44 @@
+kinesis_video_streamer:
+  ros__parameters:
+    aws_client_configuration:
+      # Specifies where you want the client to communicate. Examples include us-east-1 or us-west-2.
+      region: "us-east-1"
```
Comment (mm318) [inline simulation_ws/src/person_detection_simulation/config/kvs_config.yaml]: `cloudwatch_logs_config.yaml` and `cloudwatch_metrics_config.yaml` show `us-west-2`. Should change this to `us-west-2` to be consistent.

**Comment by mm318** [comment]
Generally looks good. I will rebase and do more testing.

**Comment by mm318** [comment]
I'm merging this into the `ros2-devel` development branch for now. I've communicated some remaining issues via e-mail.

**Pull request merged**